moped-database/migrations/1672851579949_set_fk_public_moped_proj_notes_phase_id/up.sql

```diff
@@ -0,0 +1,6 @@
+-- create foreign key relationship between moped_proj_notes and moped_phases on phase_id
+alter table "public"."moped_proj_notes"
+  add constraint "moped_proj_notes_phase_id_fkey"
+  foreign key ("phase_id")
+  references "public"."moped_phases"
+  ("phase_id") on update cascade on delete cascade;
```

Comment:
I'm wondering about the `on update cascade` and `on delete cascade` usage here, and I am not sure that want that behavior. 

I believe, as written, this says:

> When the `phase_id` of a `moped_phases` record is changed, go into `moped_proj_notes` and find all the records that point to the old ID and update them to point to the new one. 

> When a record in `moped_phases` is deleted, go into `moped_proj_notes` and delete any records that point to the being-deleted `moped_phases` record.

✅ The first behavior, the `on update` one, seems pretty great to me. I can't imagine a world when we'd be changing `phase_id`s, but it's very technically correct. 

🤔 The second behavior gives me more pause. I think we'd want the query to fail if we tried to remove a referred-to phase so we could be very intentional in how we want to handle those soon-to-be orphaned `moped_proj_notes` records. I'm pretty sure we wouldn't want to just delete them.

See next comment, for some more details, outside of this general observation.

Reply:
In practice, right now, today, and probably not forever, we're safe-ish, because we have /another/ constraint named `moped_proj_phases_moped_phases_phase_id_fk` on the `moped_proj_phases` table which maintains the relationship between the value in the `phase_id` column in `moped_proj_phases` and the FK `moped_phases(phase_id)`. 

This constraint does not have `on update` or `on delete` behavior defined, so it's this constraint that prohibits the deletion of a referred to `moped_phase` record which would cascade on to the `moped_proj_notes` table as directed by the constraint in this PR.

Reply:
My recommendation is to remove the `on delete cascade` directive so we don't cascade into deleting users comments because we remove a phase without us first hand-crafting a resolution.

I think a low priority spin-off issue would be helpful to review these constraints and make sure we've set in directives that align to what we want to delete or not and when and also to ensure that we don't have conflicting or interacting constraints that prevent each other from doing what we told them to do. The constraint in this PR enters us into that state now but ...

The whole thing is largely moot because we almost universally soft-delete records and in this case, the records that have updates and deletes cascade from them are records that are "part of the app" and not user entered data, and as a result, are very stable. 

A final thought - this is very hard for me to get all loaded into my mind and to keep straight. It's complicated! Please forgive me for what I get wrong, but FWIW, these observations are borne out of testing in my local DB.

Reply:
thanks, @frankhereford ! i fully agree and will push an update that remove `on delete cascade` from the migration. for context, i was copying the behavior of the other fkey relationship in this table, `project_id`. that may be the desired behavior in that case, but i wanted to point that out in case we want to reevaluate that.

<img width="761" alt="Screen Shot 2023-01-04 at 7 51 26 PM" src="https://user-images.githubusercontent.com/35410637/210677297-86016e7e-179b-409f-9f2f-a35bb146f308.png">